### ISSUE TITLE
azure-ts-webserver does not build

### ISSUE THREAD

**Issue by GlennPrince** [pulumi/examples#1221]
### What happened?

Trying to build the azure-ts-webserver exmple does not build due to type errors when accessing the array virtualNetwork.subnets. Full error is:

index.ts(37,23): error TS7053: Element implicitly has an 'any' type because expression of type '0' can't be used to index type 'Output<SubnetResponse[] | undefined>'.
      Property '0' does not exist on type 'Output<SubnetResponse[] | undefined>'.

### Steps to reproduce

Download example and try to run "pulumi up"

### Expected Behavior

The server would build

### Actual Behavior

The script errors

### Versions used

3.32.1

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by iwahbe** [comment]
@GlennPrince Can you give me the output of running `pulumi about` in `examples/azure-ts-webserver`. When I run `pulumi up` in `examples/azure-ts-webserver` it works fine for me.

**Comment by GlennPrince** [comment]
I have been doing some more digging and finally realized what it was. I dragged down the default template with "pulumi new azure-typescript" to get some things to work. Then I just grabbed your code file to test components which failed with the error above. 

The default new package uses strict mode only the compiler options which fails at this line:
subnet: { id: virtualNetwork.subnets[0].id },

If I turn off strict mode the code works fine. I did submit a pull request #1222 that "gets around" the strict mode failure but I'm happy to close / cancel both of these out.
